src/Shortcodes/FileLinkTracking.php

```diff
@@ -131,15 +131,21 @@ public function augmentSyncLinkTracking()
         $allFields = DataObject::getSchema()->fieldSpecs($this->owner);
         $linkedPages = [];
         $anyBroken = false;
+        $hasTrackedFields = false;
         foreach ($allFields as $field => $fieldSpec) {
             $fieldObj = $this->owner->dbObject($field);
             if ($fieldObj instanceof DBHTMLText) {
+                $hasTrackedFields = true;
                 // Merge links in this field with global list.
                 $linksInField = $this->trackLinksInField($field, $anyBroken);
                 $linkedPages = array_merge($linkedPages, $linksInField);
             }
         }
 
+        if (!$hasTrackedFields) {
+            return;
+        }
+
```

Comment:
Can use the `$linkedPages` variable as that gets populated only when there are DBHTMLText fields

```suggestion
        foreach ($allFields as $field => $fieldSpec) {
            $fieldObj = $this->owner->dbObject($field);
            if ($fieldObj instanceof DBHTMLText) {
                $hasTrackedFields = true;
                // Merge links in this field with global list.
                $linksInField = $this->trackLinksInField($field, $anyBroken);
                $linkedPages = array_merge($linkedPages, $linksInField);
            }
        }

        if (empty($linkedPages)) {
            return;
        }

```

Reply:
@michalkleiner i don't think so, because maybe you add linked page, but you don't anymore, and you want to remove them

Reply:
True, you could iterate over fields and still end up with an empty array, good call.